File: POO/temas/7/py/7.py
class Frigo:
    def __init__(self):
        self.__color = "blanco"
        self.__puerta = "cerrada"
        self.__temperatura = 20

    def abre(self):
        self.__puerta = "abierta"

    def cierra(self):
        self.__puerta = "cerrada"

    def fija_temperatura(self):
        print( "se cambia la temperatura" )

    def muestra_estado(self):
        print( f'Color: {self.__color} | Puerta {self.__puerta} | {self.__temperatura}ºC' )


mi_frigo = Frigo()
mi_frigo.muestra_estado()
mi_frigo.abre()
mi_frigo.muestra_estado()
mi_frigo.cierra()
mi_frigo.muestra_estado()

File: working/temas/c++/7.cpp
#include <iostream>
using namespace std;

class Frigo
{
    public:

    string _color;
    string _puerta;
    int _temperatura;

    Frigo()
    {
        _color = "blanco";
        _puerta = "cerrada";
        _temperatura = 20;
    }

    void abre()
    {
        _puerta = "abierta";
    }

    void cierra()
    {
        _puerta = "cerrada";
    }

    void fija_temperatura(int nueva_temperatura)
    {
        _temperatura = nueva_temperatura;
    }

    void muestra_estado()
    {
        cout << "Color: " << _color << " | Puerta " << _puerta << " | " << _temperatura << "ºC" << endl;
    }
};

int main() {
    Frigo mi_frigo;
    mi_frigo.muestra_estado();
    mi_frigo.fija_temperatura(-10);
    mi_frigo.muestra_estado();
}
